core/src/avm2/globals/flash/geom/PerspectiveProjection.as

```diff
@@ -1,44 +1,54 @@
 package flash.geom {
-    import __ruffle__.stub_constructor;
     import __ruffle__.stub_getter;
     import __ruffle__.stub_method;
     import __ruffle__.stub_setter;
+    import flash.display.DisplayObject;
     import flash.geom.Matrix3D;
     import flash.geom.Point;
 
     public class PerspectiveProjection {
-        // Getters are stubbed with what seem to be Flash's default values
+        [Ruffle(NativeAccessible)]
+        private var displayObject: DisplayObject = null;
+
+        [Ruffle(NativeAccessible)]
+        private var fov: Number = 55.0;
+
+        [Ruffle(NativeAccessible)]
+        private var center: Point = new Point(250, 250);
+
         public function PerspectiveProjection() {
-            stub_constructor("flash.geom.PerspectiveProjection");
         }
 
         public function get fieldOfView():Number {
             stub_getter("flash.geom.PerspectiveProjection", "fieldOfView");
-            return 55;
+            return this.fov;
         }
         public function set fieldOfView(value:Number) {
             stub_setter("flash.geom.PerspectiveProjection", "fieldOfView");
+            this.fov = value;
         }
 
-        public function get focalLength():Number {
-            stub_getter("flash.geom.PerspectiveProjection", "focalLength");
-            return 480.25;
-        }
-        public function set focalLength(value:Number) {
-            stub_setter("flash.geom.PerspectiveProjection", "focalLength");
-        }
+        public native function get focalLength():Number;
+
+        public native function set focalLength(value:Number);
 
         public function get projectionCenter():Point {
             stub_getter("flash.geom.PerspectiveProjection", "projectionCenter");
-            return new Point(250, 250);
+            return this.center;
         }
         public function set projectionCenter(value:Point) {
             stub_setter("flash.geom.PerspectiveProjection", "projectionCenter");
+            this.center = value;
         }
 
         public function toMatrix3D():Matrix3D {
-            stub_method("flash.geom.PerspectiveProjection", "toMatrix3D");
-            return new Matrix3D();
+            var fl: Number = this.focalLength;
+            return new Matrix3D(new <Number>[
```

Comment:
To be sure, this is a full matrix impl?

Reply:
Yes, I believe so. For other methods, especially setters are related to rendering/visual, but I believe this `toMatrix3D` just returns the Matrix3D object.

Ah, but only if the small but not very small floating point number differences are allowed. Please see `precision` in `TestToMatrix3D` test function. There is unexplainable numerical errors which is now rounded to approximation.

Except for that, I believe all features in `toMatrix3D` are implemented.